### DISCUSSION TITLE
How to patch a resource without it going OutOfSync

### DISCUSSION THREAD

**Discussion by elevy90** [argoproj/argo-cd#12026] [Q&A]
When an running argocd patch-resource on an a helm application, the application immediately goes OutOfSync and a Sync causes the change to revert. Is there a way to patch the resource so that the application doesn't go OutOfSync and the change isn't reverted upon the next Sync?

**Comment by reginaelyse** [comment]
how exactly are you trying to patch it? Could you maybe try patching it with [argocd app patch](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_patch/), [argocd app set](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_set/), or using [argocd app wait](https://argo-cd.readthedocs.io/en/stable/user-guide/commands/argocd_app_wait/)?

**Reply by elevy90** [reply]
I'm specifically trying to patch a service to change the port, which is not configurable in the helm chart. So I think argocd patch-resource is the only command that can do that, however I would have expected that if I patch it through ArgoCD, it would recognized the difference from the helm chart manifest as desired and not mark the application as out of sync, and not revert the change upon a sync.

**Comment by jaideepr97** [comment]
hi @elevy90 have you tried `ignoreDifferences`? 
https://argo-cd.readthedocs.io/en/stable/user-guide/diffing/#diffing-customization
